### PR TITLE
Add AuthController

### LINKED ISSUE #43: Authentication Controller
- [x] GetToken `auth/token`
- [x] RefreshToken `auth/refresh`
- [x] RevokeToken `auth/revoke`

### PR DESCRIPTION
**Pull request by Sjimooon**
- Add `AuthController` with enpoints (see #43).
- Add `AccountRefreshToken` entity for storing refresh tokens.
- Made corrections in `.editorconfig`.

Closes #43

### CODE REVIEW

**Review by jeongoks — DISMISSED** [review]
Looks like a very nice job done! 😎

I do, however, think that maybe @ZhakalenDk also should look through it if possible; it is approved from my side though. 😊

**Comment by ZhakalenDk** [comment]
> Looks like a very nice job done! 😎
> 
> I do, however, think that maybe @ZhakalenDk also should look through it if possible; it is approved from my side though. 😊

I'll check it out tomortow morning

File: PubHub/PubHub.API/Controllers/AuthController.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            if (account == null)
+            {
+                return Results.Problem(
+                    statusCode: UnauthorizedSpecification.STATUS_CODE,
+                    title: UnauthorizedSpecification.TITLE,
+                    detail: "Email doesn't exist.",
+                    extensions: new Dictionary<string, object?>
+                    {
+                        { "Email", email }
+                    });
```
Comment (ZhakalenDk) [inline PubHub/PubHub.API/Controllers/AuthController.cs:58]: I don't think we should inform that the email does not exist.
I'd rather have us say that either email or password didn't exist, as I think it's a security risk to expose such information.

File: PubHub/PubHub.API/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            }
+
+            // Validate user password.
+            if (!(await _userManager.CheckPasswordAsync(account, password)))
+            {
+                return Results.Problem(
+                    statusCode: UnauthorizedSpecification.STATUS_CODE,
+                    title: UnauthorizedSpecification.TITLE,
+                    detail: "Provided password is incorrect.");
```
Comment (ZhakalenDk) [inline PubHub/PubHub.API/Controllers/AuthController.cs:67]: Same here. We're telling a potential hacker that they got the account right but they're on track.

File: PubHub/PubHub.API/Controllers/AuthController.cs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            }
+
+            // Create token.
+            var tokenResult = await _authService.CreateTokenPairAsync(account);
+            if (!tokenResult.Success)
+            {
+                return Results.Problem(
+                    statusCode: InternalServerErrorSpecification.STATUS_CODE,
+                    title: InternalServerErrorSpecification.TITLE,
+                    detail: "Something went wrong and no a token couldn't be created for the given account. Please try again.",
```
Comment (ZhakalenDk) [inline PubHub/PubHub.API/Controllers/AuthController.cs:77]: Might wanna proof read this one

**Review by ZhakalenDk — CHANGES_REQUESTED** [review]

File: PubHub/PubHub.API/Controllers/AuthController.cs (hunk 1 above)
Comment (Sjimooon) [reply PubHub/PubHub.API/Controllers/AuthController.cs:58]: I'll change that to be a single check.

File: PubHub/PubHub.API/Controllers/AuthController.cs (hunk 3 above)
Comment (Sjimooon) [reply PubHub/PubHub.API/Controllers/AuthController.cs:77]: Whoops...

File: PubHub/PubHub.API/Controllers/AuthController.cs (hunk 2 above)
Comment (Sjimooon) [reply PubHub/PubHub.API/Controllers/AuthController.cs:67]: Should be fixed now 😊

**Review by ZhakalenDk — APPROVED** [review]

**Pull request merged**